### PR TITLE
Refactor valid_placement method

### PR DESCRIPTION
**Pull request by gabichuelas**

### CODE REVIEW

File: lib/board.rb [hunk 1]
```diff
@@ -31,31 +31,31 @@ def valid_coordinate?(coordinate)
 
   def valid_placement?(ship, coordinates)
```
Comment (gabichuelas) [inline lib/board.rb:30]: @amymoller I refactored the `valid_placement?` method to be cleaner/more readable.

File: lib/board.rb [hunk 2]
```diff
@@ -31,31 +31,47 @@ def valid_coordinate?(coordinate)
...
+    if valid_length && horizontally_cons && same_row
+      true
+    elsif valid_length && vertically_cons && !same_row && same_column
+      true
     else
       false
     end
+  end
 
+  def place(ship, coordinates)
```
Comment (gabichuelas) [inline lib/board.rb:70]: @amymoller I'm having trouble with this method; it's not complete, bc the tests don't pass... but when I go into pry and check whether the ship was placed on the cell object, IT HAS. So idk why it doesn't "happen" when I run the test (lines 97-99); it still thinks cell.ship == nil

File: test/board_test.rb [hunk 3]
```diff
@@ -105,10 +107,11 @@ def test_can_have_a_ship_placed_on_it
 
   def test_overlapping_ships_are_not_possible
     skip
-
+    # write extra condition in valid_placement method
+    # to check if cell.empty? == true first
```
Comment (gabichuelas) [inline test/board_test.rb]: This is just a note to us for when we get to this test.

File: lib/board.rb (hunk 1 above)
Comment (amymoller) [reply lib/board.rb:30]: Niiiice; that looks really clean!  I'm really glad we went over refactoring.  I practiced it a bit with Graham yesterday too during our session; it's so handy.

File: lib/board.rb (hunk 2 above)
Comment (gabichuelas) [reply lib/board.rb:70]: @amymoller fixed it!

**Pull request merged**